lib/stackops/lambdaRole.js

```diff
@@ -35,22 +35,26 @@ module.exports = function(currentTemplate, aliasStackTemplates, currentAliasStac
 	const role = stageStack.Resources.IamRoleLambdaExecution;
 
 	// Set role name
-	_.last(role.Properties.RoleName['Fn::Join']).push(this._alias);
+	if (role.Properties.RoleName['Fn::Join']) {
+		_.last(role.Properties.RoleName['Fn::Join']).push(this._alias);
+	}
 
 	stageStack.Resources[roleLogicalId] = stageStack.Resources.IamRoleLambdaExecution;
 	delete stageStack.Resources.IamRoleLambdaExecution;
 
 	// Replace references
 	const functions = _.filter(stageStack.Resources, ['Type', 'AWS::Lambda::Function']);
 	_.forEach(functions, func => {
-		func.Properties.Role = {
-			'Fn::GetAtt': [
-				roleLogicalId,
-				'Arn'
-			]
-		};
-		const dependencyIndex = _.indexOf(func.DependsOn, 'IamRoleLambdaExecution');
-		func.DependsOn[dependencyIndex] = roleLogicalId;
+		if (_.isEqual(func.Properties.Role, { 'Fn::GetAtt': ['IamRoleLambdaExecution', 'Arn' ]})) {
```

Comment:
I would suggest to not use `if` in this instance but use `_.filter` to select the right elements. This goes more with the functional approach, e.g.,:
`_.forEach(_.filter(functions, func => _.isEqual(func.Properties.Role, { 'Fn::Gett': ... })), func => ...)`

Reply:
got it